pkg/scheduler/framework/plugins/schedulinggates/scheduling_gates.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedulinggates
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/names"
+)
+
+// Name of the plugin used in the plugin registry and configurations.
+const Name = names.SchedulingGates
+
+// SchedulingGates checks if a Pod carries .spec.schedulingGates.
+type SchedulingGates struct {
+	enablePodSchedulingReadiness bool
+}
+
+var _ framework.EnqueuePlugin = &SchedulingGates{}
+var _ framework.EnqueueExtensions = &SchedulingGates{}
```

Comment:
Not related to this PR, but the name is a little confusing.

Reply:
Do you mean the plugin name as `SchedulingGates`?

Reply:
The name of `EnqueueExtensions`, we have `ScoreExtensions` for `ScorePlugin` for example, but the `EnqueueExtensions` is not related to plugins.

Reply:
also, should we name the plugin itself `PreEnqueue` just like PreFilter/PreScore/PreBind?

Reply:
> The name of EnqueueExtensions

Yup, it's a bit misleading as `EnqueueExtensions` is supposed to be implemented by any plugin that implements a filter-like extension point - PreFitler, Filter, Enqueue.

 > should we name the plugin itself PreEnqueue

you mean the extension point right? we may do that once we expose queuing logic as an extension point, and then rename the current one as PreEnqueue. For now, the fact is we only expose PreEnqueue in this extension point, and the core queuing logic along with the data structure is not exposed.

Reply:
> you mean the extension point right? we may do that once we expose queuing logic as an extension point, 

Harder to do that later because it will break configurations that explicitly set/unset schedulingGates.

Reply:
then let's rename it to PreEnqueue. Updated.